src/modules/interfaces.ts

```diff
@@ -337,6 +342,32 @@ export class Module {
         install: migrationClass.prototype.up,
         remove: migrationClass.prototype.down,
       };
+      const syncified = new Function(
+        'return ' +
+        migrationClass.prototype.up
+          .toString()
+          .replace(/\basync\b/g, '')
+          .replace(/\bawait\b/g, '')
+          .replace(/^/, 'function')
+          // The following are only for the test suite, but need to be included at all times
+          .replace(/\/* istanbul ignore next *\//g, '')
+          .replace(/cov_.*/g, '')
+      )();
+      const tables: string[] = [];
+      const functions: string[] = [];
+      syncified({ query: (text: string) => {
+        // TODO: Proper parsing (maybe LP?)
+        if (/^create table/i.test(text)) {
+          tables.push((text.match(/^[^"]*"([^"]*)"/) ?? [])[1]);
+        } else if (/^create or replace procedure/i.test(text)) {
+          functions.push((text.match(/^create or replace procedure ([^(]*)/i) ?? [])[0]);
+        } else if (/^create or replace function/i.test(text)) {
+          functions.push((text.match(/^create or replace function ([^(]*)/i) ?? [])[0]);
+        }
+        // Don't do anything for queries that don't match
+      }, });
```

Comment:
this makes me think that we might be able to do pull off https://github.com/iasql/docs/issues/14. right now I have to massage a bit. one of the things is remove stored procedures which are not supported by dbml/dbdocs

Reply:
This is relying on the assumption that all modules use TypeORM and they use auto-generated migration files, but that is the case right now and if we enforce the latter we could theoretically "late bind" the migrations for the modules and further reduce their size.

Reply:
oh in the dbdocs case we would actually be doing the regex on the output of `pg_dump` and it would be to remove them from the sql file

Reply:
not exactly the same, just related